exporter/elasticsearchexporter/exporter.go

```diff
@@ -158,24 +157,7 @@ func (e *elasticsearchExporter) pushLogRecord(ctx context.Context,
 	scopeSchemaUrl string) error {
 	fIndex := e.index
 	if e.dynamicIndex {
-		switch e.dynamicIndexMode {
-		case dynIdxModePrefixSuffix:
-			prefix := getFromAttributes(indexPrefix, resource, scope, record)
-			suffix := getFromAttributes(indexSuffix, resource, scope, record)
-			fIndex = fmt.Sprintf("%s%s%s", prefix, fIndex, suffix)
-		case dynIdxModeDataStream:
-			const otelSuffix = ".otel"
```

Comment:
When the output mode is otel, we still need to add the `.otel` suffix to the dataset so that the `logs-*.otel-*` index pattern matches that will be added via https://github.com/elastic/elasticsearch/pull/104455.